### PR TITLE
Delete associate

### PR DESCRIPTION
**Pull request by paphuc**
Hi team,
 
I have changed the logic to allow deleting visit, and deleting a visit will lead to delete Activities, Customers and Gifts in that Visit. This merged request also help user can select multiple labs instead of entering an input.
Please help me review this issue.

Thanks,
Phuc Pham

### CODE REVIEW

File: internal/app/api/api.go [hunk 1]
```diff
@@ -363,7 +356,7 @@ func Init(conns *InfraConns) (http.Handler, error) {
 		},
 
 		{
-			path:    "/giftassociates?cusvisitassocid={assignID:[a-z0-9-\\-]+}",
+			path:    "/giftassociates",
```
Comment (lnquy) [inline internal/app/api/api.go:359]: Bad REST API path!
Can take a look here: https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#71-url-structure

File: internal/app/api/api.go [hunk 2]
```diff
@@ -405,7 +392,7 @@ func Init(conns *InfraConns) (http.Handler, error) {
 		},
 
 		{
-			path:    "/actvisitassocs?visitid={visitID:[a-z0-9-\\-]+}",
+			path:    "/actvisitassocs",
```
Comment (lnquy) [inline internal/app/api/api.go:395]: Same as the above.

File: internal/app/api/api.go [hunk 3]
```diff
@@ -447,7 +428,7 @@ func Init(conns *InfraConns) (http.Handler, error) {
 		},
 
 		{
-			path:    "/cusvisitassocs?visitid={visitID:[a-z0-9-\\-]+}",
+			path:    "/cusvisitassocs",
```
Comment (lnquy) [inline internal/app/api/api.go:431]: Same as the above.

File: internal/app/api/handler/actvisitassoc/act_visit_assoc.go [hunk 4]
```diff
@@ -50,22 +50,19 @@ func (h *Handler) Get(w http.ResponseWriter, r *http.Request) {
 
 // GetAll handle get all activity associates HTTP Request
 func (h *Handler) GetAll(w http.ResponseWriter, r *http.Request) {
```
Comment (lnquy) [inline internal/app/api/handler/actvisitassoc/act_visit_assoc.go]: Should change func name as its behavior changed, right?

File: internal/app/api/handler/actvisitassoc/act_visit_assoc.go [hunk 5]
```diff
@@ -123,7 +120,8 @@ func (h *Handler) Delete(w http.ResponseWriter, r *http.Request) {
 
 // DeleteByVisitID handle delete activity associate HTTP Request
 func (h *Handler) DeleteByVisitID(w http.ResponseWriter, r *http.Request) {
-	if err := h.srv.DeleteByVisitID(r.Context(), mux.Vars(r)["visitID"]); err != nil {
+	queryValues := r.URL.Query()
```
Comment (lnquy) [inline internal/app/api/handler/actvisitassoc/act_visit_assoc.go]: Should get query argument and validate first before calling service.

Comment (lnquy) [inline internal/app/api/handler/actvisitassoc/act_visit_assoc.go]: Should get query argument and validate first before calling service.
Same for other places.

File: internal/app/cusvisitassoc/cus_visit_assoc.go [hunk 6]
```diff
@@ -77,15 +76,20 @@ func (s *Service) Update(ctx context.Context, cusVisitAssoc types.CusVisitAssoc)
...
-	return errors.New("Can not delete this customer because it has some assigned gifts.")
 }
 
 // DeleteByVisitID a customer visit associate
 func (s *Service) DeleteByVisitID(ctx context.Context, visitID string) error {
-	return s.repo.DeleteByVisitID(ctx, visitID)
+	if err := s.assocRepo.DeleteByCusVisitAssocID(ctx, visitID); err == nil {
+		return s.repo.DeleteByVisitID(ctx, visitID)
+	} else {
+		return err
```
Comment (lnquy) [inline internal/app/cusvisitassoc/cus_visit_assoc.go:89]: Remove this `else`.
Should omit `else` in your code as much as possible as it make your code nested on another level.

File: internal/app/visit/visit.go [hunk 7]
```diff
@@ -81,8 +86,13 @@ func (s *Service) Update(ctx context.Context, visit types.Visit) error {
...
 func (s *Service) Delete(ctx context.Context, id string) error {
-	if !s.assocCusRepo.IsAssignedCustomer(ctx, "", id) && !s.assocActRepo.IsAssignedActivity(ctx, "", id) {
-		return s.repo.Delete(ctx, id)
+	if err := s.assocActService.DeleteByVisitID(ctx, id); err == nil {
+		if err = s.assocCusService.DeleteByVisitID(ctx, id); err == nil {
+			return s.repo.Delete(ctx, id)
+		} else {
+			return err
+		}
+	} else {
```
Comment (lnquy) [inline internal/app/visit/visit.go]: Cleanup these line of codes.
Remove as much `else` as possible.

File: web/src/components/activity/ActivityListByVisit.vue [hunk 8]
```diff
@@ -105,6 +105,7 @@ export default {
   mounted() {
     getActVisitAssocsByVisitID(this.visit.id)
       .then(resp => {
+        console.log(resp.data)
```
Comment (lnquy) [inline web/src/components/activity/ActivityListByVisit.vue]: Should remove debug log when you finished the development.

**Review by pthethanh — APPROVED** [review]

**Pull request merged**